### PR TITLE
Add anchors to bullets

### PR DESCRIPTION
**Pull request by corroded**
As discussed, here are all the anchors to each bullet point. I am a bit sleepy though (12am here haha) so this will need your eyes, @christopheradams . Also I was unsure how to name the anchors of some so it can be pretty long sometimes.

Let me know your thoughts and if I need to change something!

### CODE REVIEW

**Comment by christopheradams** [comment]
This is awesome!

Before merging though I should update the markdown linter rules to work with this so the checks pass.

**Comment by corroded** [comment]
Ah yes. I was thinking about that but was too tired to check how to fix it. Should it be in a separate branch or here?

**Comment by christopheradams** [comment]
We can do it here, so we can make sure the checks pass.

**Comment by corroded** [comment]
Got it. I did a quick scan and fixed the inline HTML one by excluding the rule. The problem now is we have 

`README.md:66: MD023 Headers must start at the beginning of the line`

and 

`README.md:61: MD032 Lists should be surrounded by blank lines`

Not sure how you want to fix those. For MD032, I can just add a blank line after each <a> tag but weird that MD023 pops up now. Probably because the presence of the HTML tags are effing up the parser? 

for reference, line 66 where MD023 blows up is a comment in an elixir code block and it parses it as a header instead, complaining that it should not be indented.

 Let me know how you want to fix it :)

**Comment by christopheradams** [comment]
I think this formatting might solve both the list spacing rule and avoid that headers bug (which I've encountered before):

```
* <a name="spaces-indentation"></a>
  Use two **spaces** per indentation level.
  No hard tabs.
  <sup>[[link](#spaces-indentation)]</sup>
```

**Comment by corroded** [comment]
You're right, that works! Added a couple of bullet points I missed last night. Tested the links and they're all good so far.

File: README.md [hunk 1]
```diff
@@ -435,27 +484,33 @@ If you're looking for other projects to contribute to please see the
...
   ```elixir
   def cool?(var) do
     # Complex check if var is cool not possible in a pure function.
   end
   ```
 
-* Private functions with the same name as public functions should start with
+* <a name="private-functions-with-same-name-as-public"></a>
+  Private functions with the same name as public functions should start with
+  <sup>[[link](#private-functions-with-same-name-as-public)]</sup>
```
Comment (christopheradams) [inline README.md]: this line needs to be moved down one

File: README.md [hunk 2]
```diff
@@ -804,7 +923,9 @@ directives (see [Modules](#modules)).
 
 ### Exceptions
 
-* Make exception names end with a trailing `Error`.
+* <a name="exception-names"></a>
+  Make exception names end with a trailing `Error`.
+  <sup>[[link](#excepton-names)]</sup>
```
Comment (christopheradams) [inline README.md]: typo should be `#exception-names`

File: README.md [hunk 3]
```diff
@@ -822,8 +943,10 @@ directives (see [Modules](#modules)).
   end
   ```
 
-* Use lowercase error messages when raising exceptions, with no trailing
+* <a name="lowercase-error-messages"></a>
+  Use lowercase error messages when raising exceptions, with no trailing
   punctuation.
+  <sup>[[link](#)]</sup>
```
Comment (christopheradams) [inline README.md]: should be `#lowercase-error-messages`

Comment (corroded) [inline README.md]: good spots! will fix these asap

**Pull request merged**